tests/phpunit/CRM/Civicase/Service/CaseCategoryCustomFieldExtendsTest.php

```diff
@@ -0,0 +1,146 @@
+    $this->assertEquals($customFieldCreated['values'][0]['label'], $label);
+  }
+
+  /**
+   * Test that calling twice the create method does not produce an error.
+   */
+  public function testIsNotCreatedTwice() {
+    $entityValue = 'Test Entity Value';
+    $label = 'Test Label';
+    $entityTypeFunction = 'Test Entity Type Function';
+    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
+
+    // First call.
+    $caseCategoryCustomFieldExtends->create($entityValue, $label, $entityTypeFunction);
+    // Second call.
+    $caseCategoryCustomFieldExtends->create($entityValue, $label, $entityTypeFunction);
+
+    $customFieldCreated = $this->getCustomFieldByValue($entityValue);
+    $this->assertEquals(1, $customFieldCreated['count']);
+  }
+
+  /**
+   * Test deleting an existing custom field.
+   */
+  public function testIsSuccessfullyDeleted() {
+    // Create a random instance of custom field.
+    $caseTypeCategory = $this->createRandomCustomField();
+    // We verify that is correctly stored.
+    $customField = $this->getCustomFieldByValue($caseTypeCategory['value']);
+    $this->assertEquals(1, $customField['count']);
+
+    // Delete the custom field.
+    (new CaseCategoryCustomFieldExtends())->delete($caseTypeCategory['value']);
+
+    // And verify is not counted anymore.
+    $customField = $this->getCustomFieldByValue($caseTypeCategory['value']);
+    $this->assertEquals(0, $customField['count']);
+  }
+
+  /**
+   * Test that calling twice the delete method does not produce an error.
+   */
+  public function testDeleteCanBeCalledTwice() {
+    // Create a random instance of custom field.
+    $caseTypeCategory = $this->createRandomCustomField();
+    // We verify that is correctly stored.
+    $customField = $this->getCustomFieldByValue($caseTypeCategory['value']);
+    $this->assertEquals(1, $customField['count']);
+
+    $caseCategoryCustomFieldExtends = new CaseCategoryCustomFieldExtends();
+    // First call to delete method.
+    $caseCategoryCustomFieldExtends->delete($caseTypeCategory['value']);
+    // Second call to delete method.
+    $caseCategoryCustomFieldExtends->delete($caseTypeCategory['value']);
+
+    // And verify is not counted anymore.
+    $customField = $this->getCustomFieldByValue($caseTypeCategory['value']);
+    $this->assertEquals(0, $customField['count']);
+  }
+
+  /**
+   * Returns a custom field by it value.
+   *
+   * @param string $value
+   *   Value of the custom field extends.
+   *
+   * @return array
+   *   Custom Field Extends details
+   */
+  private function getCustomFieldByValue($value) {
+    return civicrm_api3('OptionValue', 'get', [
+      'sequential' => 1,
+      'value' => $value,
+      'option_group_id' => 'cg_extend_objects',
+    ]);
+  }
+
+  /**
+   * Create a random instance of custom field extends.
+   *
+   * @return array
+   *   Custom Field Extends details
+   */
+  private function createRandomCustomField() {
```

Comment:
Can we rename this function to reflect what it does. Maybe `CreateOptionValueForCgExtendsGroup` or something better. This does not actually create a custom field. A custom field is created using the CustomField entity API.

Reply:
Thanks for the notes! I was clearly confused with the meaning of the instances created.
I renamed everything based on the pre-existing method of the service class being tested: `getCgExtendOptionValue`. That's why I called the instances `cgExtendOptionValue`, and followed the same logic for other parts.
Let me know if this requires some other fixes.